### PR TITLE
[NTOS:MM] Properly set PTE dirty bit after serving an access fault.

### PR DESCRIPTION
**Pull request by zefklop**
## Purpose

Set the dirty bit. The CPU is actually writing to the page, but it doesn't set the bit.

JIRA issue: [CORE-17595](https://jira.reactos.org/browse/CORE-17595)

### CODE REVIEW

File: ntoskrnl/mm/i386/page.c [hunk 1]
```diff
@@ -814,14 +814,24 @@ MmSetDirtyBit(PEPROCESS Process, PVOID Address, BOOLEAN Bit)
...
 
-    MiLockProcessWorkingSetUnsafe(Process, PsGetCurrentThread());
+        MiLockProcessWorkingSetUnsafe(Process, PsGetCurrentThread());
 
-    MiMakePdeExistAndMakeValid(MiAddressToPde(Address), Process, MM_NOIRQL);
+        MiMakePdeExistAndMakeValid(MiAddressToPde(Address), Process, MM_NOIRQL);
+    }
 
     PointerPte = MiAddressToPte(Address);
     // We shouldnl't set dirty bit on non-mapped adresses
```
Comment (HBelusca) [inline ntoskrnl/mm/i386/page.c:837]:
```suggestion
    // We shouldn't set dirty bit on non-mapped addresses
```

**Review by HBelusca — APPROVED** [review]

**Review by JoachimHenze — CHANGES_REQUESTED** [review]
I tested PR3748 on top of 0.4.15-dev-2707-gd042f51, but it did not fix the bug for me: Again I got the strong name bug on first try.
I can not judge whether the patch is an improvement or not, but it definitely does not fix CORE-17595 and therefore at least the commit comment would need adjustment.
Committing makes only sense in case it would fix "something else".

**Comment by JoachimHenze** [comment]
No sorry, this does still not work. I got the strong-name-bug CORE-17595 again upon first try. It does not fix the ticket. I attached the fresh logs to the ticket
https://jira.reactos.org/browse/CORE-17595?focusedCommentId=128626&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-128626

**Comment by zefklop** [comment]
Closing this one, reopening with a propoer description

**Pull request closed without merging**